### PR TITLE
Email template

### LINKED ISSUE #17: Email template
I can't read the e-mail if it just show the json for rule violations like a sausage.

### PR DESCRIPTION
**Pull request by leocwolter**
Closes #17

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -1,24 +1,31 @@
...
 		resp += chunk;
 	});
 
 	response.on('end', function () {
 		var pullRequests = JSON.parse(resp);
-		console.log(pullRequests.length);
-	});
+		pullRequests = pullRequests.filter(function(item) {return item["pull_request"]});
+		var violations = validator.validate(pullRequests, rules);
+		mailer.send(violations);
```
Comment (williammizuta) [inline app.js]: bad indentation

File: app.js [hunk 2]
```diff
@@ -1,24 +1,31 @@
...
 
 https.request(options, function(response) {
-	var resp = ''
+	var resp = '';
 	response.on('data', function (chunk) {
 		resp += chunk;
 	});
 
 	response.on('end', function () {
 		var pullRequests = JSON.parse(resp);
```
Comment (williammizuta) [inline app.js]: It is strange a variable named `pullRequests` contains more than pull requests. I think we should apply the filter function here or change the variable name.

File: app.js [hunk 3]
```diff
@@ -1,24 +1,31 @@
+var util = require('util');
 var https = require('https');
 var fs = require('fs');
 var config = JSON.parse(fs.readFileSync('config.json', 'utf8'));
+var rules = JSON.parse(fs.readFileSync('rules.json', 'utf-8'));
+var global = JSON.parse(fs.readFileSync('global_rules.json', 'utf-8'));
+var validator = require('./app/validator.js')(global);
```
Comment (williammizuta) [inline app.js]: Why pull request rules and global rules are called in different ways in validator?

File: global_rules.json [hunk 4]
```diff
@@ -0,0 +1,6 @@
+{
+	"pull-requests-limit" : {
+		"value": 20,
+		"message": "Exceeded pull request limit"
+	}
+}
```
Comment (williammizuta) [inline global_rules.json]: This file should not be commited. Only the sample one.

File: rules.json [hunk 5]
```diff
@@ -0,0 +1,9 @@
+[
+	{
+		"create" : {
+			"value": 5,
+			"message" : "Too old"
+		},
+		"update" : 2
+	}
+]
```
Comment (williammizuta) [inline rules.json]: This file should not be commited. Only the sample one.

File: app/pimp_my_rule.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+				return rule;
+			},
+			getMessage: function(strategy){
+				if(typeof(rule) == "object") {
+					return rule.message;
+				}
+				return "The following pull request violated the rule: " + strategy;
+			}
+		}
+	return pimpedRule;
```
Comment (williammizuta) [inline app/pimp_my_rule.js]: bad indentation

File: app/mailer.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+var config = JSON.parse(fs.readFileSync('config.json', 'utf8'));
+var template = fs.readFileSync('resources/template.ejs', 'utf-8');
+
+var transporter = nodemailer.createTransport({
+    service: config.mail.service,
+    auth: {
+        user: config.mail.user,
+        pass: config.mail.password
+    }
+});
```
Comment (williammizuta) [inline app/mailer.js]: bad indentation

File: app.js (hunk 3 above)
Comment (leocwolter) [inline app.js]: Because global rules will be always de same for every validator.validate call, unlike the other rules

File: app.js (hunk 2 above)
Comment (leocwolter) [inline app.js]: :ok:

File: app.js (hunk 1 above)
Comment (leocwolter) [inline app.js]: It's ok at sublime

File: app/mailer.js (hunk 7 above)
Comment (leocwolter) [inline app/mailer.js]: :ok_hand:

File: app/pimp_my_rule.js (hunk 6 above)
Comment (leocwolter) [inline app/pimp_my_rule.js]: :ok_hand:

File: global_rules.json (hunk 4 above)
Comment (leocwolter) [inline global_rules.json]: :ok_hand:

**Pull request merged**